src/common/src/session_config/mod.rs

```diff
@@ -12,1082 +12,284 @@
+    /// Sets the application name to be reported in statistics and logs.
+    /// See <https://www.postgresql.org/docs/14/runtime-config-logging.html#:~:text=What%20to%20Log-,application_name,-(string)>
+    #[parameter(default = "", flags = "REPORT")]
+    application_name: String,
 
+    /// It is typically set by an application upon connection to the server.
     /// see <https://www.postgresql.org/docs/current/runtime-config-client.html#GUC-DATESTYLE>
-    date_style: DateStyle,
+    #[parameter(default = "", rename = "DATESTYLE")]
+    date_style: String,
 
-    /// To force the usage of lookup join instead of hash join in batch execution
-    batch_enable_lookup_join: BatchEnableLookupJoin,
+    /// Force the use of lookup join instead of hash join when possible for local batch execution.
+    #[parameter(default = true, rename = "RW_BATCH_ENABLE_LOOKUP_JOIN")]
+    batch_enable_lookup_join: bool,
 
-    /// To open the usage of sortAgg instead of hash agg when order property is satisfied in batch
+    /// Enable usage of sortAgg instead of hash agg when order property is satisfied in batch
     /// execution
-    batch_enable_sort_agg: BatchEnableSortAgg,
+    #[parameter(default = true, rename = "RW_BATCH_ENABLE_SORT_AGG")]
+    batch_enable_sort_agg: bool,
 
-    /// It's the max gap allowed to transform small range scan scan into multi point lookup.
-    max_split_range_gap: MaxSplitRangeGap,
+    /// The max gap allowed to transform small range scan scan into multi point lookup.
+    #[parameter(default = 8)]
+    max_split_range_gap: i32,
 
-    /// see <https://www.postgresql.org/docs/14/runtime-config-client.html#GUC-SEARCH-PATH>
+    /// Sets the order in which schemas are searched when an object (table, data type, function, etc.)
+    /// is referenced by a simple name with no schema specified.
+    /// See <https://www.postgresql.org/docs/14/runtime-config-client.html#GUC-SEARCH-PATH>
+    #[parameter(default = SearchPath::default())]
     search_path: SearchPath,
 
     /// If `VISIBILITY_MODE` is all, we will support querying data without checkpoint.
+    #[parameter(default = VisibilityMode::default())]
     visibility_mode: VisibilityMode,
 
-    /// see <https://www.postgresql.org/docs/current/transaction-iso.html>
+    /// See <https://www.postgresql.org/docs/current/transaction-iso.html>
+    #[parameter(default = IsolationLevel::default())]
     transaction_isolation_level: IsolationLevel,
 
-    /// select as of specific epoch
-    query_epoch: QueryEpoch,
+    /// Select as of specific epoch.
+    /// Sets the historical epoch for querying data. If 0, querying latest data.
+    #[parameter(default = ConfigNonZeroU64::default())]
+    query_epoch: ConfigNonZeroU64,
 
     /// Session timezone. Defaults to UTC.
-    #[educe(Default(expression = "ConfigString::<TIMEZONE>(String::from(\"UTC\"))"))]
-    timezone: Timezone,
+    #[parameter(default = "UTC", check_hook = check_timezone)]
+    timezone: String,
```

Comment:
What about putting a `Tz` here and do the check in `FromStr`?

Reply:
I would suggest abandoning the `check_hook` design from Postgres' GUC. With a more modern programming language, we can do the check during the parsing and return error there.

In other words, by default we call `FromStr::from_str` when setting the config from the provided string. This can be overridden by specifying a `parse_fn = ..` here. If the developers want to customize the parsing or checking, they can...

- either define a newtype and implement `FromStr` there,
- or specifying a `parse_fn` in the attributes.

Reply:
Maybe we can provide a override parser?

Reply:
I think `Timezone` should be treated as an `enum`, and be parsed and validated by `FromStr`, like `VisibilityMode`. Since this is not introduced in this PR, may do it later.

Reply:
After careful consideration, I realize the `check_hook` might have a use case for checking integer parameters. For example, a block size can be 1024, 2048 but cannot be 1 (too small) or 99999 (too big) or 2050 (not a multiple of 512). IIRC MySQL use something like `max, min, step` to describe this, but `check_hook` can also handle this easily.

Reply:
> > check_hook might have a use case for checking integer parameter
> 
> Maybe we can add an `allowed_value`?

Can you explain more on the idea? `allowed_value` sounds like a list of allowed values, but here you need a function to perform the check.

Reply:
`allowed_value ` can be used on enum values, but not a good practice for other checks. @BugenZhao mentioned we can perform the check by implementing it in `parse_fn`.